hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableFactory.java

```diff
@@ -132,7 +133,7 @@ private void sanityCheck(Configuration conf, ResolvedSchema schema) {
       }
 
       Arrays.stream(recordKeys)
-          .filter(field -> !fields.contains(field))
+          .filter(field -> !fields.contains(getRootLevelFieldName(field)))
           .findAny()
```

Comment:
Do we support nested primary key also in this patch ?

Reply:
Uhm, from what i understand, Spark-SQL supports nested primaryKey and precombineField.

The changes I made here are to standardize the validations in Spark and Flink.
https://issues.apache.org/jira/browse/HUDI-4051
https://github.com/apache/hudi/pull/5517/files